src/components/presentations/NewArticle/NewArticle.jsx

```diff
@@ -1,53 +1,124 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import { EditorState, convertToRaw } from 'draft-js';
+import { toast } from 'react-toastify';
+import draftToHtml from 'draftjs-to-html';
+import uploadToCloudnary from '../../../utils/uploadToCloudnary';
 import NewArticleForm from './NewArticleForm/NewArticleForm';
+import validateImage from '../../../utils/validateImage';
 import NavBar from '../../shared/NavBar/NavBar';
 
 class NewArticle extends Component {
   constructor() {
     super();
-    this.state = {};
+    this.state = {
+      category: '',
+      keywords: [],
+      imageUrl: '',
+      editorState: EditorState.createEmpty(),
+      body: '',
+      title: '',
+      abstract: '',
+    };
+    this.processInput = this.processInput.bind(this);
+    this.onChange = this.onChange.bind(this);
+    this.onEditorStateChange = this.onEditorStateChange.bind(this);
   }
 
-  render() {
-    const {
-      handleInputChange,
-      handleImageChange,
-      handleEditorState,
-      handleSubmit,
-      handleTextAreaChange,
-      title,
-      content,
-      textarea,
-    } = this.props;
+  onChange(e) {
+    this.setState({ [e.target.name]: e.target.value.toLowerCase() });
+  }
+
+  onEditorStateChange = editorState => {
+    const body = draftToHtml(convertToRaw(editorState.getCurrentContent()));
+    this.setState({
+      body,
+      editorState,
+    });
+  };
+
+  saveToCloudinary = async e => {
+    const form = new FormData();
+    const imageData = e.target.files[0];
+    const validFormat = validateImage(imageData);
+    if (validFormat.valid) {
+      toast.info(validFormat.message, {
+        type: toast.TYPE.INFO,
+        closeButton: false,
+        position: toast.POSITION.TOP_CENTER,
+      });
+      form.append('file', imageData);
+      const res = await uploadToCloudnary(form);
+      this.setState({ imageUrl: res.url });
+      toast.dismiss();
+    } else {
+      toast.error(validFormat.message);
+      e.target.value = '';
+    }
+  };
+
+  saveKeywords = e => {
+    e.preventDefault();
+    const keywordData = e.target.attributes.getNamedItem('name').value;
+    const { keywords } = this.state;
+    this.setState({ keywords: [...keywords, keywordData] });
+  };
 
+  saveCategory = e => {
+    e.preventDefault();
+    const categoryData = e.target.attributes.getNamedItem('name').value;
+    this.setState({ category: categoryData });
+  };
+
+  processInput = async (e, isDraft) => {
```

Comment:
Use a better name that reflects what this does maybe something like `saveOrPublish`...